### DISCUSSION TITLE
I can't share my view

### DISCUSSION THREAD

**Discussion by inukaze** [epoptes/epoptes#85] [Q&A]
Hi there, yesterday on a classroom on the academy i work, the teacher try to share the view from him monitor to the client machines just show the message : 

VNC Viewer Information
Error : No Matching security types 

How i can fix it ?

**Comment by inukaze** [comment]
Ok i can't edit. well all machines are using Lubuntu 16.04 LTS 32 Bits. i try with this on server and client sides :

settings set org.gnome.Vino require-encryption false
gsettings set org.gnome.Vino require-encryption false

But don't work, of anyway, i first try just from server side, after from client side, after from both sides, and the result ever is the same.

**Comment by alkisg** [comment]
Hi, which epoptes version are you using? The latest ones are these:
```
dpkg -l '*epoptes*' | grep ^ii
ii  epoptes        1.0.0-1+t201809210732~ubuntu18.04.1 all          Computer lab management tool
ii  epoptes-client 1.0.0-1+t201809210732~ubuntu18.04.1 all          Computer lab management tool (client)
```

**Comment by inukaze** [comment]
On teacher machines : 

```
$ sudo lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial

estudiante ~ $ dpkg -l '*epoptes*' | grep ^ii
ii  epoptes        0.5.10-1     all          Computer lab management tool
```

On students machines : 
```
$ sudo lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial

dpkg -l '*epoptes*' | grep ^ii
ii  epoptes-client        0.5.10-1     all          Computer lab management tool
```

When i try share the host screen with clients on fullscreen appears the error dialog : 
VNC Viewer Information
Error : No Matching security types

**Comment by alkisg** [comment]
Please update to Epoptes 1.0 by using the PPA:
http://www.epoptes.org/documentation/ppa

Also, Epoptes doesn't use "vino" etc, so if you installed it as a service, disable it. Epoptes is using x11vnc.

**Comment by inukaze** [comment]
i am trying right now. i has added the ppa, like explain in this parte > http://www.epoptes.org/documentation/ppa

i notice epoptes 1.0 is not available for Ubuntu 16.04 on the ppa ( https://launchpad.net/~epoptes/+archive/ubuntu/ppa?field.series_filter=xenial ) 

ok i will go try manually upgrade.

**Comment by alkisg** [comment]
It's not in the stable PPA for xenial because it hasn't been tested much there.
It's in the proposed PPA:
https://code.launchpad.net/~epoptes/+archive/ubuntu/proposed/+packages

**Comment by inukaze** [comment]
Thanks for the information. well i upgrade, i when i try run

```
$ epoptes 

** (epoptes:23366): WARNING **: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
Traceback (most recent call last):
  File "/usr/bin/epoptes", line 99, in <module>
    main()
  File "/usr/bin/epoptes", line 89, in main
    egui = gui.EpoptesGui()
  File "/usr/lib/python3/dist-packages/epoptes/ui/gui.py", line 83, in __init__
    self.get('mi_remote_assistance').set_property('visible', False)
AttributeError: 'NoneType' object has no attribute 'set_property'

```

**Comment by inukaze** [comment]
Ok on the that ppa package you need put dependency "epoptes-client" package. that error gave me, because i don't had installed epoptes-client package.

**Comment by alkisg** [comment]
In all PPAs and all distros, epoptes recommends epoptes-client.
There are only a few broken distros like linuxmint that don't install recommended packages by default, breaking debian policy.

**Comment by inukaze** [comment]
Well i had upgraded to epoptes 1.0 , but the clients machines dont appears like "on". i had upgraded the cert file too. with "epoptes-client -c" and reboot the machines. and still all icons on gray like off. but i had turn on all machines.

How i fix it?

**Comment by alkisg** [comment]
Run epoptes on the server from a terminal, and epoptes-client on the client from a terminal, and paste the output.

**Comment by inukaze** [comment]
Server :

```
 $ epoptes

** (epoptes:15076): WARNING **: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
Got clients: None
```

Client : 
```
$ epoptes-client
Cannot resolve 192.168.2.6 while DNS appears to work, aborting.
```

**Comment by alkisg** [comment]
Could you try putting the following line in the client's /etc/hosts?
```192.168.2.6 server```

Explanation: usually we setup epoptes-client by puting that in /etc/hosts, but I think that you put "SERVER=192.168.2.6" in /etc/default/epoptes-client instead. And this triggered a bug in the new Epoptes version, which needs to be resolved, i.e. if SERVER is an ip, to use it without waiting for DNS.

**Comment by inukaze** [comment]
Yes i am using /etc/default/default/epoptes-client and the line "SEVER=192.168.2.6"

Ok i had commented on /etc/default/epoptes-client
#SERVER=192.168.2.6

i add to /etc/hosts
192.168.2.6 server

Well when i try to share my view, i get the same message :

**VNC Viewer Information
"Error : No Matching security types"**

**Comment by alkisg** [comment]
I'm guessing this comes from vino?
Epoptes doesn't use vino, either disable or uninstall it.
